c_src/ex_dtls/dtls.c

```diff
@@ -89,10 +89,32 @@ unsigned char *export_keying_material(SSL *ssl) {
   int res = SSL_export_keying_material(ssl, material, len,
                                        "EXTRACTOR-dtls_srtp", 19, NULL, 0, 0);
   if (res != 1) {
-    return NULL;
-  }
-
-  return material;
+    free(material);
+    return NULL;
+  }
+  DEBUG("Keying material %s", material);
+
+  KeyingMaterial *keying_material;
+  keying_material = (KeyingMaterial *)malloc(sizeof(KeyingMaterial));
+  memset(keying_material, 0, sizeof(KeyingMaterial));
+  keying_material->client =
+      (unsigned char *)malloc(len / 2 * sizeof(unsigned char));
+  keying_material->server =
+      (unsigned char *)malloc(len / 2 * sizeof(unsigned char));
+  memset(keying_material->client, 0, len / 2);
+  memset(keying_material->server, 0, len / 2);
+
+  memcpy(keying_material->client, material, master_key_len);
+  memcpy(keying_material->server, material + master_key_len, master_key_len);
+  memcpy(keying_material->client + master_key_len,
+         material + 2 * master_key_len, master_salt_len);
+  memcpy(keying_material->server + master_key_len,
+         material + 2 * master_key_len + master_salt_len, master_salt_len);
```

Comment:
if you introduced some 'index' variable and incremented it on each memcpy, it would be more readable